ui/src/linking/linking-utils.ts

```diff
@@ -91,9 +95,13 @@ export function getLocationTracksForJointConnections(
 }
 
 export function getSuggestedSwitchId(suggestedSwitch: SuggestedSwitch): string {
-    return `${suggestedSwitch.geometrySwitchId}_${suggestedSwitch.alignmentEndPoint
-        ?.locationTrackId}_${suggestedSwitch.joints.map(
-        (j) => j.number + j.matches.map((m) => m.locationTrackId + m.segmentIndex + m.m),
+    return `${suggestedSwitch.geometrySwitchId}_${
+        suggestedSwitch.alignmentEndPoint?.locationTrackId
+    }_${Object.entries(suggestedSwitch.trackLinks).map(
+        ([t, js]) =>
+            t +
+            js.segmentJoints.map((j) => j.number + j.segmentIndex + j.m) +
+            (js.topologyJoint ? js.topologyJoint.number + js.topologyJoint.trackEnd : ''),
```

Comment:
Nämä `.map()`:it voisi erottaa vaikka omiin funktioihinsa. Nyt tässä alkaa olla sen verran paljon sisäkkäistä looppausta ja akronyymimuuttujia, että multa ainakin katoaa todella nopeasti käsitys siitä minkälaisia id:itä tämä oikein generoi